### PR TITLE
refactor: Consolidate duplicated ItemGrid library components

### LINKED ISSUE #298: Consolidate duplicated code across `ItemGrid` library components
# ItemGrid Architecture Analysis
### Current Component Structure

The codebase currently has THREE separate components for displaying library grids:

1. **`ItemGrid`** (components/ItemGrid/ItemGrid.bs/.xml) - 900 lines
   - **Extends**: JRGroup
   - **Purpose**: Generic library grid for TV shows, collections, live TV, boxsets, genres, etc.
   - **Special handling**: Music and Movie collection types have conditional logic

2. **`MovieLibraryView`** (components/ItemGrid/MovieLibraryView.bs/.xml) - 972 lines  
   - **Extends**: JRScreen
   - **Purpose**: Dedicated movie library viewer
   - **Key features**:
     - Movie-specific presentation views (Movies, MoviesGrid, Studios, Genres)
     - Movie poster with metadata overlay (year, runtime, rating, critic rating, overview)
     - Backdrop image management
     - Advanced filter options (Official Ratings, Years, Genres, Studios)
     - Logo loading for movies

3. **`MusicLibraryView`** (components/ItemGrid/MusicLibraryView.bs/.xml) - 821 lines
   - **Extends**: JRScreen  
   - **Purpose**: Dedicated music library viewer
   - **Key features**:
     - Music-specific presentation views (Artists Presentation/Grid, Album Artists, Albums, Genres)
     - Artist metadata display (song count, album count, genres)
     - Backdrop image management
     - Simpler filter options (All, Favorites)

---

### What Was Duplicated

By examining the code, I found the previous developer copy/pasted ~70-80% of the base `ItemGrid` component code to create the specialized views. Here's the breakdown:

### Duplicated Code (~700 lines each):

1. **Core initialization logic**
   - Node setup (`setupNodes()`)
   - Task initialization (`LoadItemsTask2`, logo tasks)
   - Data structures (`m.data`, `m.genreData`)
   - Grid configuration
   - Voice box setup
   - Alpha menu handling

2. **Event handlers** (nearly identical):
   - `onItemFocused()`
   - `onItemSelected()`
   - `onvoiceFilter()`
   - `alphaSelectedChanged()`
   - `optionsClosed()`
   - `ItemDataLoaded()`
   - `loadMoreData()`
   - Key event handling

3. **Display settings management**:
   - `getLibraryDisplaySetting()` calls
   - `setLibraryDisplaySetting()` calls
   - Sort, filter, and view persistence

4. **Backdrop management**:
   - `SetBackground()` function
   - `newBGLoaded()` handler
   - Background swap animation

### Unique/Customized Code (~200-300 lines each):

1. **`MovieLibraryView` specific**:
   - `setMoviesOptions()` - Movie view/sort/filter options
   - `FilterDataLoaded()` - Advanced filter handling (genres, studios, years, ratings)
   - Movie metadata display logic (critic ratings, community ratings, runtime, etc.)
   - Movie logo loading
   - `GetFiltersTask` integration

2. **`MusicLibraryView` specific**:
   - `setMusicOptions()` - Music view/sort/filter options  
   - Artist metadata display (album count, song count, genres)
   - Artist logo loading
   - Simpler filter options

3. **`ItemGrid` specific**:
   - TV show handling
   - Live TV guide integration
   - Collection/boxset handling
   - Generic fallback logic

---

### ✅ Pros

1. **Clear separation of concerns**
   - Each library type has its own dedicated component
   - Easy to find movie-specific or music-specific code
   - No risk of breaking one library when modifying another

2. **Customized UI per library type**
   - Movie library can show rich metadata (critic ratings, runtime, overview)
   - Music library can show artist-specific data (album count, song count)
   - Each has tailored view options matching the content type

3. **Performance isolation**
   - Movie filter loading (genres, studios, years) doesn't impact music
   - Each component only loads what it needs
   - No conditional checks for library type at runtime in many places

4. **Easier to understand for newcomers**
   - Self-contained components are easier to read
   - No massive if/else chains for different library types
   - Clear entry points for each library type

5. **Simpler testing**
   - Can test movie library independently
   - Can test music library independently  
   - Fewer edge cases per component

---

### ❌ Cons

1. **Massive code duplication** (~700 lines × 2 = 1,400 lines)
   - Bug fixes need to be applied 3× (ItemGrid, MovieLibraryView, MusicLibraryView)
   - Feature additions need to be implemented 3×
   - Refactoring efforts are tripled
   - High risk of drift - one component gets updated, others don't

2. **Maintenance nightmare**
   - Noticed a bug in pagination? Fix it in 3 places
   - Want to improve voice search? Update 3 files
   - Need to add a new sort option? Modify 3 components
   - Display settings migration needed updating all 3

3. **Inconsistent behavior risk**
   - Alpha menu works differently in different views
   - Voice search might behave inconsistently
   - Options dialog might have subtle differences
   - Loading spinners might not match

4. **Difficult to add new library types**
   - Want to add a Books library? Copy/paste again
   - Want to add a Podcasts library? 1,000 more lines
   - Pattern encourages continued duplication

5. **Testing burden**
   - Need to test identical functionality 3× 
   - Unit tests need to be duplicated
   - Integration tests need to cover all variants

6. **Memory footprint**
   - Three separate components in memory
   - Duplicate code in package
   - Larger app size

### PR DESCRIPTION
**Pull request by cewert**
## Summary
Consolidates three duplicated library grid components (ItemGrid, MovieLibraryView, MusicLibraryView) into a single unified BaseGridView component using the presenter pattern.

- Movie and Music libraries now show the overhang i.e. Library title and the clock
- Movie and Music libraries now support the app-wide backdropFader component 
- Standardizes (and fixes) the item poster sizes in movie libraries (6 columns)
- Changes the music item poster sizes (5 columns)
- Removes both of the presentation views from the music library and migrates deprecated settings as needed

Fixes #298

### New Components
- **BaseGridView**: Unified library grid component with presenter pattern
- **GridPresenterBase**: Base class defining presenter interface
- **MoviePresenter**: Movie-specific views and metadata display
- **MusicPresenter**: Music-specific views and artist/album handling
- **TVShowPresenter**: TV show-specific views and series handling
- **GenericPresenter**: Fallback for LiveTV, Collections, Boxsets, etc.
- **LibraryViewFactory**: Centralized factory for creating library views
- **PresentationBackdrop**: Right-half screen backdrop for presentation views

### Removed Components
- ItemGrid (899 lines)
- MovieLibraryView (1,024 lines)
- MusicLibraryView (858 lines)
- Orphaned factory functions in ShowScenes.bs

## Benefits
- ✅ Eliminates ~1,400 lines of duplicated code
- ✅ Single source of truth for grid functionality
- ✅ Easier to maintain and extend with new media types
- ✅ Consistent behavior across all library types
- ✅ Cleaner separation of concerns via presenter pattern

Before:
before-grid
After:
after-grid

Before:
before-presentation
After:
after-presentation

Before:
before-music

After:
after-music

### CODE REVIEW

File: source/GridView/MoviePresenter.bs [hunk 1]
```diff
@@ -0,0 +1,437 @@
...
+        genreNode.bold = true
+        movieGenresGroup.appendChild(genreNode)
+      end if
+    end if
+
+    ' Dynamically populate movieDescription with tagline and overview
+    if isValid(movieDescriptionGroup)
+      ' Clear existing children
+      movieDescriptionGroup.removeChildrenIndex(movieDescriptionGroup.getChildCount(), 0)
+
```
[the remark this answers is not in the document]
Comment (cewert) [reply source/GridView/MoviePresenter.bs]: wrong wrong https://developer.roku.com/docs/references/brightscript/interfaces/ifsgnodechildren.md#removechildrenindexnum_children-as-integer-index-as-integer--as-boolean
